lib/internal/validators.js

```diff
@@ -142,11 +143,24 @@ const validateBuffer = hideStackFrames((buffer, name = 'buffer') => {
   }
 });
 
+function validateEncoding(data, encoding) {
+  const normalizedEncoding = normalizeEncoding(encoding);
+  const length = data.length;
+
+  if (normalizedEncoding === 'hex' && length % 2 !== 0) {
+    throw new ERR_INVALID_ARG_VALUE('encoding', encoding,
+                                    `is invalid for data of length ${length}`);
+  }
+
+  // TODO(bnoordhuis) Add BASE64 check?
```

Comment:
Should this be added?

Reply:
@mocllina If I understand correctly, currently, that check does not happen, so while ideally it ought to, I don't think its absence needs to hold this up. I imagine a concern is that it may be an expensive check? /ping @bnoordhuis

Reply:
That's correct. The current check is O(1), checking it's valid base64 is O(n).